### ISSUE TITLE
update docs and annotations to standard build

### ISSUE THREAD

**Issue by mscottnelson** [nexus-js/ui#109]
Currently the documentation process involves some manual editing of code, and valid JSDoc annotations aren't necessarily picked up by the generated documentation. The primary reason for this is that there's no easy way to annotate full-fledged examples using JSDoc annotations alone.

I recommend setting up a separation of concerns around tutorials/examples and JSDoc annotations, with a means to bind them together during documentation generation. This will allow documention.js to generate appropriate annotations even if whoever is maintaining the documentation doesn't anticipate a particular JSDoc usage in the code base (eg. if multiple @examples are specified they are not currently rendered, default parameters are currently not rendered, etc), leaving that for documentation.js to sort out.

An example putting this into practice with documentation.js handling code annotation with additional example/tutorial pages pulled into the final documentation of the API can be found in this project:
https://github.com/mapbox/mapbox-gl-js/tree/master/docs

The upside of doing this will be examples and tutorials that can live independently and be as complex as desired, while keeping code as clean and self-documenting as possible. Additionally, documentation will not require as much manual "tweaking" when changes are made to the code-base.

**Comment by taylorbf** [comment]
I understand these concerns. It is an idiosyncratic setup and is difficult for others to contribute to, yes. 

I spent a surprising amount of time looking at options for documentation... I didn't find any that satisfied the needs of the project. The main reason for this is that different components in the lib are best communicated in different ways. A dial interface might do fine to be described by a regular JSDoc-style API. But something like the [Matrix](https://nexus-js.github.io/ui/api/#matrix) constructor and others are really poorly described by an API. Yes, you can do it, but I think it is *much* better introduced by a sort of walk-through. In that sense it's not exactly a 'tutorial' -- which is the word I use for it, too -- but its more like a narrative API. 

I think it's important to keep the narrative/tutorial pages (tuning, timing, counter, sequence, matrix, etc). in the same space as the rest of the API. If we separate API and these 'tutorials', I think one of them invariably becomes dominant and the other is ignored. .. in this case I'm sure that the API, containing the majority of the information, would be the sole reference, really.

So, basically, I'm hesitant to overhaul. I don't expect that much further development of the library in the foreseeable future. The docs could be made better, absolutely. but I'd opt for cleaning up and expanding upon the current setup, and/or putting time into writing better descriptions in the current docs.

If we want to make a separate page with *actual* tutorials / longer examples, that is something I would totally support, although would only be able to contribute to in a limited capacity.

**Comment by mscottnelson** [comment]
Fair enough. Thanks for the explanation!
